prereise/gather/demanddata/transportation_electrification/data_process.py

```diff
@@ -0,0 +1,132 @@
+import pandas as pd
+
+#Download CSV of NHTS 2017 from https://nhts.ornl.gov/
+
+def data_filtering(census_division, filepath: str = "trippub.csv"):
+    """Filter raw NHTS data to be used in mileage.py
+    
+    :param str filepath: filepath for the "trippub.csv" file in downloaded NHTS data
+    :return: (*pd.DataFrame*) -- filtered and sorted trip data
+    """
+
+    nhts_raw = pd.read_csv(filepath)
+    
+    #filter to be only vehicle trips (TRPTRANS values 1-6)
+    nhts = nhts_raw[nhts_raw.TRPTRANS.isin(range(1,7))]
+    
+    #filter out repeated trips (VMT = -1)
+    nhts = nhts[nhts.TRPMILES != -1]
+    
+    #get correct census division
+    nhts = nhts[nhts.CENSUS_D == census_division]
+
+    #dataframe for use built here
+    #columns taken directly from original csv:
+    sorted_data = pd.DataFrame({"Household": nhts["HOUSEID"], 
+                   "Vehicle ID": nhts["VEHID"], 
+                   "Person ID": nhts["PERSONID"],
+                   "Scaling Factor Applied": nhts["WTTRDFIN"],
+                   "Trip Number": nhts["TDTRPNUM"],
+                   "Date": nhts["TDAYDATE"], 
+                   "Day of Week": nhts["TRAVDAY"],              
+                   "If Weekend": nhts["TDWKND"],                
+                   "Trip start time (HHMM)": nhts["STRTTIME"],
+                   "Trip end time (HHMM)": nhts["ENDTIME"],
+                   "Travel Minutes": nhts["TRVLCMIN"],
+                   "Dwell time": nhts["DWELTIME"],
+                   "Miles traveled": nhts["VMT_MILE"],          
+                   "Vehicle miles travelled": nhts["TRPMILES"], 
+                   "why from": nhts["WHYFROM"],
+                   "why to": nhts["WHYTO"],
+                   "Vehicle type": nhts["TRPTRANS"],            
+                   "Household vehicle count": nhts["HHVEHCNT"],
+                   "Household size": nhts["HHSIZE"]
+                   })
+    
+    #columns that require calculations: 21-28
+    sorted_data["Start time (hour decimal)"] = [row//100 + row % 100 / 60 for row in sorted_data["Trip start time (HHMM)"]]
+    sorted_data["End time (hour decimal)"] = [row//100 + row % 100 / 60 for row in sorted_data["Trip end time (HHMM)"]]
+    sorted_data['Dwell time (hour decimal)'] = ''
```

Comment:
If all we're doing here is expanding the set of columns without filling meaningful data, then instead of:
```python
df["a"] = '',
df["b"] = '',
...
```
we can add multiple columns at once without them being set to string/object type with something like:
```python
df = df.reindex([df.columns + ["a", "b", ...], axis="columns")
```